Installation issue: ADIOS2

### Steps to reproduce the issue

ADIOS2  v2.10 and above is missing lower bound on HDF5 (1.12) where for H5VLpublic.h is defined
```console
 -   adios2@2.11.0~aws+blosc2+bzip2~campaign~cuda~dataspaces+fortran+hdf5~ipo~kokkos+libcatalyst~libpressio+mgard+mpi~pic+png~python~rocm+sst~sycl+sz+xrootd+zfp build_system=cmake build_type=Release generator=make patches:=c721484 platform=linux os=ubuntu24.04 target=skylake %c,cxx,fortran=gcc@13.3.0
[+]      ^bzip2@1.0.8~debug~pic+shared build_system=generic platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^diffutils@3.12 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]              ^libiconv@1.18 build_system=autotools libs:=shared,static platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]      ^c-blosc2@2.15.1+avx2~ipo+lizard+lz4+snappy+zlib+zstd build_system=cmake build_type=Release generator=make platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^lizard@2.1 build_system=makefile platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^lz4@1.10.0+pic build_system=makefile libs:=shared,static platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]          ^snappy@1.2.1~ipo+pic+shared build_system=cmake build_type=Release generator=make platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]          ^zlib-ng@2.3.2+compat~new_strategies+opt+pic+shared build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]          ^zstd@1.5.7+programs build_system=makefile compression:=none libs:=shared,static platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]      ^cmake@3.31.11~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]          ^curl@8.18.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=openssl platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]              ^nghttp2@1.67.1 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]          ^ncurses@6.5-20250705~symlinks+termlib abi=none build_system=autotools patches:=7a351bc platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]      ^compiler-wrapper@1.0 build_system=generic platform=linux os=ubuntu24.04 target=skylake
[e]      ^gcc@13.3.0+binutils+bootstrap~graphite+libsanitizer~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' platform=linux os=ubuntu24.04 target=x86_64
[+]      ^gcc-runtime@13.3.0 build_system=generic platform=linux os=ubuntu24.04 target=skylake
[e]      ^glibc@2.39 build_system=autotools platform=linux os=ubuntu24.04 target=x86_64
[+]      ^gmake@4.4.1~guile build_system=generic platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]      ^hdf5@1.8.23~cxx~fortran~hl~ipo+mpi+shared~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make patches:=f42732a platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]      ^libcatalyst@2.0.0~fortran~ipo~mpi~python build_system=cmake build_type=Release generator=make platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]          ^conduit@0.9.5~adios+blt_find_mpi~caliper~doc~doxygen+examples+fortran+hdf5+hdf5_compat~ipo+mpi+parmetis~python+shared~silo+test+utilities~zfp build_system=cmake build_type=Release generator=make platform=linux os=ubuntu24.04 target=skylake %c,cxx,fortran=gcc@13.3.0
[+]              ^blt@0.7.1 build_system=generic patches:=116702b platform=linux os=ubuntu24.04 target=skylake %c,cxx,fortran=gcc@13.3.0
[+]              ^metis@5.1.0~gdb~int64~ipo~no_warning~real64+shared build_system=cmake build_type=Release generator=make patches:=4991da9,93a7903,b1225da platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]              ^parmetis@4.0.3~gdb~int64~ipo+shared build_system=cmake build_type=Release generator=make patches:=4f89253,50ed208,704b84f platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]      ^libfabric@2.4.0~asan~cuda~debug~kdreg~level_zero~lsan~rocm~tsan~ubsan~uring build_system=autotools fabrics:=sockets,tcp,udp platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]      ^libffi@3.5.2 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]      ^libpng@1.6.55~ipo~pic build_system=cmake build_type=Release generator=make libs:=shared,static platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]      ^mgard@1.6.0~cuda~ipo+openmp~rocm+serial~timing~unstructured build_system=cmake build_type=Release generator=make patches:=1c64ad2 platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]          ^libarchive@3.8.5+iconv build_system=autotools compression:=bz2lib,lz4,lzma,lzo2,zlib,zstd crypto=openssl libs:=shared,static programs:=none xar=libxml2 platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]              ^lzo@2.10 build_system=autotools libs:=shared,static platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^protobuf@33.1~ipo+shared build_system=cmake build_type=Release generator=make platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]              ^abseil-cpp@20240722.0~ipo+shared build_system=cmake build_type=Release cxxstd=17 generator=make platform=linux os=ubuntu24.04 target=skylake %cxx=gcc@13.3.0
[+]          ^python@3.14.3+bz2+ctypes+dbm~debug~freethreading+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~static~tests~tkinter+uuid+zlib+zstd build_system=generic platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]              ^expat@2.7.3+libbsd build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]                  ^libbsd@0.12.2 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]                      ^libmd@1.1.0 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]              ^gdbm@1.25 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]              ^gettext@1.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]                  ^tar@1.35 build_system=autotools zip=pigz platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]                      ^pigz@2.8 build_system=makefile platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]              ^sqlite@3.50.4+column_metadata+fts+rtree build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^sed@4.9 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^yaml-cpp@0.8.0~ipo+pic+shared~tests build_system=cmake build_type=Release generator=make patches:=0bb42be platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]      ^openmpi@5.0.9+atomics~cuda~debug+fortran~gpfs~internal-hwloc~internal-libevent~internal-pmix~ipv6~java~lustre~memchecker~openshmem~rocm~romio+rsh~static~two_level_namespace+vt+wrapper-rpath build_system=autotools fabrics:=none romio-filesystem:=none schedulers:=none platform=linux os=ubuntu24.04 target=skylake %c,cxx,fortran=gcc@13.3.0
[+]          ^autoconf@2.72 build_system=autotools platform=linux os=ubuntu24.04 target=skylake
[+]              ^m4@1.4.20+sigsegv build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]                  ^libsigsegv@2.14 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^automake@1.18.1 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^hwloc@2.12.2~cairo~cuda~gl~level_zero~libudev+libxml2~nvml~opencl+pci~rocm build_system=autotools libs:=shared,static platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]              ^libpciaccess@0.17 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]                  ^util-macros@1.20.1 build_system=autotools platform=linux os=ubuntu24.04 target=skylake
[+]          ^libevent@2.1.12+openssl build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^libtool@2.4.7 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]              ^findutils@4.10.0 build_system=autotools patches:=440b954 platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^numactl@2.0.19 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^openssh@10.2p1+gssapi build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]              ^krb5@1.22.2+shared build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]                  ^bison@3.8.2~color build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]              ^libedit@3.1-20240808 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^perl@5.42.0+cpanm+opcode+open+shared+threads build_system=generic platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]              ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]          ^pmix@6.0.0~munge~python build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^prrte@4.0.0 build_system=autotools schedulers:=none platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]              ^flex@2.6.3+lex~nls build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]      ^pkgconf@2.5.1 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]      ^sz@2.1.12.5~examples~fortran~hdf5~ipo~netcdf~openmp~pastri~python~random_access+shared~stats~time_compression build_system=cmake build_type=Release generator=make platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]      ^xrootd@5.9.1~client_only~davix+ec+http~ipo~krb5~python+readline~scitokens-cpp build_system=cmake build_type=Release cxxstd=17 generator=make patches:=0d03eed platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]          ^isa-l@2.31.1 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]              ^nasm@2.16.03 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]              ^yasm@1.3.0 build_system=autotools patches:=8e0d7a1 platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^json-c@0.18~ipo build_system=cmake build_type=Release generator=make platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^libxcrypt@4.4.38~obsolete_api build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^libxml2@2.15.1+pic~python+shared build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^openssl@3.6.1~docs+shared build_system=generic certs=mozilla platform=linux os=ubuntu24.04 target=skylake %c,cxx=gcc@13.3.0
[+]              ^ca-certificates-mozilla@2025-08-12 build_system=generic platform=linux os=ubuntu24.04 target=skylake
[+]          ^readline@8.3 build_system=autotools patches:=21f0a03 platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^util-linux-uuid@2.41 build_system=autotools platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]          ^xz@5.6.3~pic build_system=autotools libs:=shared,static platform=linux os=ubuntu24.04 target=skylake %c=gcc@13.3.0
[+]      ^zfp@0.5.5~aligned~c~cuda~fasthash~fortran~ipo~openmp~profile~python+shared~strided~twoway+utilities bsws=64 build_system=cmake build_type=Release generator=make platform=linux os=ubuntu24.04 target=skylake %c,cxx,fortran=gcc@13.3.0
```

### Error message

```bash
=> No binary for adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy found: installing from source
==> Installing adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy [57/57]
==> Using cached archive: /home/dokken/Documents/src/spack/var/spack/cache/_source-cache/archive/0a/0a2bd745e3f39745f07587e4a5f92d72f12fa0e2be305e7957bdceda03735dbf.tar.gz
==> Using cached archive: /home/dokken/Documents/src/spack/var/spack/cache/_source-cache/archive/c7/c7214845bc9e4262deb901f9d689236e014f5193018617675bea4bed80ca20aa
==> Applied patch https://github.com/ornladios/ADIOS2/commit/0bdda7d4729b898397e024010b1e82cb72921501.patch?full_index=1
==> adios2: Executing phase: 'cmake'
==> adios2: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/home/dokken/Documents/src/spack/opt/spack/linux-skylake/gmake-4.4.1-5rizl5hhwim4j2mhbjc5kbwqpjagyohc/bin/make' '-j2'

4 errors found in build log:
     12887    cd /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu && /home/dokken/Documents/src/spack/opt/spack/linux-skylake/cmake
              -3.31.11-hktjuxhljv7staa7oullkpjzktxdtq4q/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/sp
              ack-src /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-src/source/h5vol /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6eu
              icnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/source/h5vol /tmp/dokk
              en/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/source/h5vol/CMakeFiles/adios2_h5vol.dir/DependInfo.cmake "--color="
     12888    make[2]: Leaving directory '/tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu'
     12889    /home/dokken/Documents/src/spack/opt/spack/linux-skylake/gmake-4.4.1-5rizl5hhwim4j2mhbjc5kbwqpjagyohc/bin/make  -f source/h5vol/CMakeFiles/adios2_h5vol.dir/build.make sour
              ce/h5vol/CMakeFiles/adios2_h5vol.dir/build
     12890    make[2]: Entering directory '/tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu'
     12891    [ 85%] Building C object source/h5vol/CMakeFiles/adios2_h5vol.dir/H5VolReadWrite.c.o
     12892    cd /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/source/h5vol && /home/dokken/Documents/src/spack/opt/spack/linux-
              skylake/compiler-wrapper-1.0-i54t7tjn3prjyb363kdjgrkiawikdvyu/libexec/spack/gcc/gcc -DADIOS2_USE_MPI -DH5_BUILT_AS_DYNAMIC_LIB -DHAVE_UNISTD_H -DMPICH_SKIP_MPICXX -DOMPI_S
              KIP_MPICXX -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200809L -Dadios2_h5vol_EXPORTS -I/tmp/dokken/spack-stage/spack-
              stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-src/bindings/C -I/tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-src/so
              urce -I/tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/source -isystem /home/dokken/Documents/src/spack/opt/spack/li
              nux-skylake/hdf5-1.8.23-mxqof6btvnwhfmhbtf5aaw24xite3loe/include -isystem /home/dokken/Documents/src/spack/opt/spack/linux-skylake/openmpi-5.0.9-3sjbojbuyrro4xqkir23hyx2wv
              3c2lo7/include -Wno-format-zero-length -O3 -DNDEBUG -std=gnu99 -fPIC -MD -MT source/h5vol/CMakeFiles/adios2_h5vol.dir/H5VolReadWrite.c.o -MF CMakeFiles/adios2_h5vol.dir/H5
              VolReadWrite.c.o.d -o CMakeFiles/adios2_h5vol.dir/H5VolReadWrite.c.o -c /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-src/source
              /h5vol/H5VolReadWrite.c
  >> 12893    /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-src/source/h5vol/H5VolReadWrite.c:1:10: fatal error: H5VLpublic.h: No such file or
               directory
     12894        1 | #include "H5VLpublic.h"
     12895          |          ^~~~~~~~~~~~~~
     12896    compilation terminated.
  >> 12897    make[2]: *** [source/h5vol/CMakeFiles/adios2_h5vol.dir/build.make:82: source/h5vol/CMakeFiles/adios2_h5vol.dir/H5VolReadWrite.c.o] Error 1
     12898    make[2]: Leaving directory '/tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu'
  >> 12899    make[1]: *** [CMakeFiles/Makefile2:1991: source/h5vol/CMakeFiles/adios2_h5vol.dir/all] Error 2
     12900    make[1]: *** Waiting for unfinished jobs....
     12901    [ 86%] Building CXX object bindings/Fortran/CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_attribute.cpp.o
     12902    cd /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/bindings/Fortran && /home/dokken/Documents/src/spack/opt/spack/li
              nux-skylake/compiler-wrapper-1.0-i54t7tjn3prjyb363kdjgrkiawikdvyu/libexec/spack/gcc/g++ -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX -Dadios2_fortran_EXPORTS -I/tmp/dokken/spack
              -stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/bindings/Fortran -I/tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtp
              b23yg6nu42yh5smy/spack-src/bindings/C -I/tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-src/source -I/tmp/dokken/spack-stage/spack
              -stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/source -O3 -DNDEBUG -std=gnu++14 -fPIC -MD -MT bindings/Fortran/CMakeFiles/adios2_fortran.dir/f2c
              /adios2_f2c_attribute.cpp.o -MF CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_attribute.cpp.o.d -o CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_attribute.cpp.o -c /tmp/dokk
              en/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-src/bindings/Fortran/f2c/adios2_f2c_attribute.cpp
     12903    [ 86%] Building CXX object bindings/Fortran/CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_io.cpp.o
     12904    cd /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/bindings/Fortran && /home/dokken/Documents/src/spack/opt/spack/li
              nux-skylake/compiler-wrapper-1.0-i54t7tjn3prjyb363kdjgrkiawikdvyu/libexec/spack/gcc/g++ -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX -Dadios2_fortran_EXPORTS -I/tmp/dokken/spack
              -stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/bindings/Fortran -I/tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtp
              b23yg6nu42yh5smy/spack-src/bindings/C -I/tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-src/source -I/tmp/dokken/spack-stage/spack
              -stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/source -O3 -DNDEBUG -std=gnu++14 -fPIC -MD -MT bindings/Fortran/CMakeFiles/adios2_fortran.dir/f2c
              /adios2_f2c_io.cpp.o -MF CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_io.cpp.o.d -o CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_io.cpp.o -c /tmp/dokken/spack-stage/spack-
              stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-src/bindings/Fortran/f2c/adios2_f2c_io.cpp
     12905    [ 86%] Building CXX object bindings/Fortran/CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_variable.cpp.o

     ...

     13000    cd /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/bindings/Fortran && /home/dokken/Documents/src/spack/opt/spack/li
              nux-skylake/cmake-3.31.11-hktjuxhljv7staa7oullkpjzktxdtq4q/bin/cmake -E cmake_link_script CMakeFiles/adios2_fortran.dir/link.txt --verbose=1
     13001    /home/dokken/Documents/src/spack/opt/spack/linux-skylake/compiler-wrapper-1.0-i54t7tjn3prjyb363kdjgrkiawikdvyu/libexec/spack/gcc/g++ -fPIC -O3 -DNDEBUG -Wl,--dependency-fi
              le=CMakeFiles/adios2_fortran.dir/link.d -shared -Wl,-soname,libadios2_fortran.so.2.11 -o ../../lib/libadios2_fortran.so.2.11.0 CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c
              _adios.cpp.o CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_attribute.cpp.o CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_io.cpp.o CMakeFiles/adios2_fortran.dir/f2c/adios2_f2
              c_variable.cpp.o CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_engine.cpp.o CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_operator.cpp.o CMakeFiles/adios2_fortran.dir/module
              s/adios2_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_functions_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_functions_allocate_mod.f90.o CMakeFiles/ad
              ios2_fortran.dir/modules/adios2_parameters_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_adios_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_adios_init_m
              od.F90.o CMakeFiles/adios2_fortran.dir/modules/adios2_adios_init_serial_smod.F90.o CMakeFiles/adios2_fortran.dir/modules/adios2_attribute_mod.f90.o CMakeFiles/adios2_fortr
              an.dir/modules/adios2_attribute_data_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_io_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_io_open_mod.F90.o CMa
              keFiles/adios2_fortran.dir/modules/adios2_io_open_serial_smod.F90.o CMakeFiles/adios2_fortran.dir/modules/adios2_io_define_variable_mod.f90.o CMakeFiles/adios2_fortran.dir
              /modules/adios2_io_define_derived_variable_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_io_define_attribute_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios
              2_engine_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_engine_begin_step_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_engine_put_mod.f90.o CMakeFiles/ad
              ios2_fortran.dir/modules/adios2_engine_get_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_variable_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_variable_
              min_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_variable_max_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_operator_mod.f90.o  -Wl,-rpath,/tmp/dokken/s
              pack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skylake/c-blosc2-2.15.1-55tv
              7zdapyepyf4mvi6b6sgxqbxcbeqr/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skylake/lz4-1.10.0-ukrn5xeauml7sntkzb6sb7luqamhjw25/lib:/home/dokken/Documents/src/spack/
              opt/spack/linux-skylake/zlib-ng-2.3.2-ken3wsofeerqpfgzg5dqy3t2ngy65xz4/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skylake/zfp-0.5.5-sshdvwisrgdpsjr742gddjgbyjzcw
              tyi/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skylake/mgard-1.6.0-aguy6gqryjdls7x7vq4w2lklj5vgwdvw/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skylake/
              protobuf-33.1-ocpwgfzqz4h3zptjl53h6ghyyw2gk6dh/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skylake/abseil-cpp-20240722.0-z22pfr5hmghewitwxiaccfer6del7k5r/lib:/hom
              e/dokken/Documents/src/spack/opt/spack/linux-skylake/zstd-1.5.7-g4ssqphslaomxsoov7vq5csl26zbdplm/lib:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
              ::::::::::::::::::::::::::::::::::::::::::::::::::::::: ../../lib/libadios2_c.so.2.11.0 ../../lib/libadios2_core.so.2.11.0 -lgfortran -lquadmath -Wl,-rpath-link,/home/dokk
              en/Documents/src/spack/opt/spack/linux-skylake/c-blosc2-2.15.1-55tv7zdapyepyf4mvi6b6sgxqbxcbeqr/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skylake/lz4-1.10.0-ukr
              n5xeauml7sntkzb6sb7luqamhjw25/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skylake/zlib-ng-2.3.2-ken3wsofeerqpfgzg5dqy3t2ngy65xz4/lib:/home/dokken/Documents/src/sp
              ack/opt/spack/linux-skylake/zfp-0.5.5-sshdvwisrgdpsjr742gddjgbyjzcwtyi/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skylake/mgard-1.6.0-aguy6gqryjdls7x7vq4w2lklj5v
              gwdvw/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skylake/protobuf-33.1-ocpwgfzqz4h3zptjl53h6ghyyw2gk6dh/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skyl
              ake/abseil-cpp-20240722.0-z22pfr5hmghewitwxiaccfer6del7k5r/lib:/home/dokken/Documents/src/spack/opt/spack/linux-skylake/zstd-1.5.7-g4ssqphslaomxsoov7vq5csl26zbdplm/lib
     13002    cd /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu/bindings/Fortran && /home/dokken/Documents/src/spack/opt/spack/li
              nux-skylake/cmake-3.31.11-hktjuxhljv7staa7oullkpjzktxdtq4q/bin/cmake -E cmake_symlink_library ../../lib/libadios2_fortran.so.2.11.0 ../../lib/libadios2_fortran.so.2.11 ../
              ../lib/libadios2_fortran.so
     13003    make[2]: Leaving directory '/tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu'
     13004    [ 92%] Built target adios2_fortran
     13005    make[1]: Leaving directory '/tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-x6qz6eu'
  >> 13006    make: *** [Makefile:139: all] Error 2

See build log for details:
  /tmp/dokken/spack-stage/spack-stage-adios2-2.11.0-x6qz6euicnb2nqtpb23yg6nu42yh5smy/spack-build-out.txt
```

### Information on your system

* **Spack:** 1.2.0.dev0 (https://github.com/spack/spack/commit/34eba77df2992389b4a36ed521d035433179a52d)
* **Builtin repo:** https://github.com/spack/spack-packages/commit/63fe6392a29184c3769f3dd9800253faab66b89f
* **Python:** 3.12.3
* **Platform:** linux-ubuntu24.04-skylake


### Additional information

@ax3l  @vicentebolea  @williamfgc
[spack-build-out.txt](https://github.com/user-attachments/files/25572449/spack-build-out.txt)
[spack-build-env.txt](https://github.com/user-attachments/files/25572450/spack-build-env.txt)

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/adios2/package.py
+++ b/repos/spack_repo/builtin/packages/adios2/package.py
@@ -185,6 +185,7 @@ class Adios2(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("dataspaces@1.8.0:", when="+dataspaces")
 
     depends_on("hdf5@:1.12", when="@:2.8 +hdf5")
+    depends_on("hdf5@1.12:", when="@2.9: +hdf5")
     depends_on("hdf5~mpi", when="+hdf5~mpi")
     depends_on("hdf5+mpi", when="+hdf5+mpi")
 

